packages/rfw/README.md

```diff
@@ -208,7 +328,409 @@ class _ExampleState extends State<Example> {
+      onTapDown: source.handler<GestureTapDownCallback>( // this returns a function that takes a TapDownDetails
+        ['onTapDown'],
+        (HandlerTrigger trigger) { // "trigger" is the function that will send the event to RemoteWidget.onEvent
+          return (TapDownDetails details) { // this is the function that is returned by handler() above
+            trigger(); // the function calls "trigger"
+          };
+        },
+      ),
+      child: source.optionalChild(['child']),
+    );
+```
+
+In some cases, the arguments sent to the callback (the
+`TapDownDetails` in this case) are useful and should be passed to the
+`onEvent` as part of its arguments. This can be done by passing some
+values to the `trigger` method, as in:
+
+```dart
+    return GestureDetector(
+      onTapDown: source.handler(['onTapDown'], (HandlerTrigger trigger) {
+        return (TapDownDetails details) => trigger({
+          x: details.globalPosition.dx,
+          y: details.globalPosition.dy,
+        });
+      }),
+      child: source.optionalChild(['child']),
+    );
+```
+
+Any arguments in the `event` get merged with the arguments passed to
+the trigger.
+
+### Developing remote widget libraries
+
+Remote widget libraries are usually defined using a Remote Flutter
+Widgets text library file (`rfwtxt` extension), which is then compiled
+into a binary library file (`rfw` extension) on the server before
+being sent to the client.
+
+The format of text library files is defined in detail in the API
+documentation of the
+[`parseLibraryFile`](https://pub.dev/documentation/rfw/latest/formats/parseLibraryFile.html)
+function.
+
+Compiling a text `rfwtxt` file to the binary `rfw` format can be done
+by calling
+[`encodeLibraryBlob`](https://pub.dev/documentation/rfw/latest/formats/encodeLibraryBlob.html)
+on the results of calling `parseLibraryFile`.
+
+The example in `example/wasm` has some [elaborate remote
+widgets](https://github.com/flutter/packages/blob/main/packages/rfw/example/wasm/logic/calculator.rfwtxt),
+including some that manipulate state (`Button`).
+
+#### Lists
+
+Let us consider a remote widget library that is used to render data in
+this form and is registered with the `Runtime` under the name
+`server`:
+
+```
+{ games: [
+{"rating": 8.219, "users-rated": 16860, "name": "Twilight Struggle", "rank": 1, "link": "/boardgame/12333/twilight-struggle", "id": 12333},
+{"rating": 8.093, "users-rated": 11750, "name": "Through the Ages: A Story of Civilization", "rank": 2, "link": "/boardgame/25613/through-ages-story-civilization", "id": 25613},
+{"rating": 8.088, "users-rated": 34745, "name": "Agricola", "rank": 3, "link": "/boardgame/31260/agricola", "id": 31260},
+{"rating": 8.082, "users-rated": 8913, "name": "Terra Mystica", "rank": 4, "link": "/boardgame/120677/terra-mystica", "id": 120677},
+...
+```
+
+(This is both valid JSON and a valid RFW data file, which shows how
+similar the two syntaxes are.)
```

Comment:
Probably clearest if this mentions the defined term "configuration data" from above, and/or the name `DynamicContent` from the API, to pin down the phrase "RFW data file" as corresponding to that and not e.g. to the file format for remote widget libraries that was discussed just above.

(Or perhaps that should happen at "render data in this form" just above.)

Reply:
added a bunch of prose and links here